server/server_test.go

```diff
@@ -81,3 +86,169 @@ func forwardSignal(t *testing.T, outbound chan<- struct{}, inbound <-chan os.Sig
 	}
 	outbound <- s
 }
+
+func TestNewSocketServer(t *testing.T) {
+	const (
+		func_name   = "TestNewSocketServer"
+		mock_socket = "socket_addr:socket_port"
+	)
+	cfg := new(config.Configuration)
+	cfg.UnixSocketName = mock_socket
+
+	mock_server := &http.Server{
+		Addr:         cfg.UnixSocketName,
+		Handler:      nil,
+		ReadTimeout:  15 * time.Second,
+		WriteTimeout: 15 * time.Second,
+	}
+
+	if ret := newSocketServer(cfg, nil); ret != nil {
+		switch {
+		case ret.Addr != mock_server.Addr:
+			t.Errorf("[%s] Addr invalide: %v != %v",
+				func_name, ret.Addr, mock_server.Addr)
+			fallthrough
+		case ret.ReadTimeout != mock_server.ReadTimeout:
+			t.Errorf("[%s] ReadTimeout invalide: %v != %v",
+				func_name, ret.ReadTimeout, mock_server.ReadTimeout)
+			fallthrough
+		case ret.WriteTimeout != mock_server.WriteTimeout:
+			t.Errorf("[%s] WriteTimeout invalide: %v != %v",
+				func_name, ret.WriteTimeout, mock_server.WriteTimeout)
+		}
+		ret.Close()
+	} else {
+		t.Errorf("[%s] ret is Nil", func_name)
+	}
+}
+
+func TestNewMainServer_(t *testing.T) {
+	const (
+		func_name    = "TestNewMainServer_"
```

Comment:
Is there a reason why this name has an underscore at the end? Is it possible to remove this?